fastlane/lanes/build.rb

```diff
@@ -173,7 +169,7 @@
     create_release(
       repository: GITHUB_REPO,
       version: version,
-      release_notes_file_path: File.join(PROJECT_ROOT_FOLDER, 'WordPress', 'Resources', 'release_notes.txt'),
+      release_notes_file_path: WORDPRESS_RELEASE_NOTES_PATH,
       release_assets: archive_zip_path.to_s,
       prerelease: options[:beta_release]
     )
```

Comment:
Unrelated to this PR, but while reviewing this PR it jumped out to me that WordPress and Jetpack are still unbalanced on those steps:
 - We do the `create_release` dance for WordPress (here in `build_and_upload_app_store_connect`, but not for Jetpack (in `build_and_upload_jetpack_for_app_store`)
    - I guess it makes sense to only create one single GitHub Release for _both_ WP and JP, instead of one for each, so that's likely why it's like that
    - But it still makes things oddly unbalanced in each of the lane's implementation. And why should this GitHub Release creation have any more reason to be supported by the WP lane rather than the JP one?
    - So I think it would probably make more sense to split the GitHub Release creation into its own, separate lane, and call it separately when appropriately needed, rather than providing `create_release:true` as part of one of those lanes
 - The lanes for WP and JP have different names for the same thing (`build_and_upload_app_store_connect` vs `build_and_upload_jetpack_for_app_store`)
    - I think it would make sense, especially once you'll have separated the GitHub Release creation into its own lane and that both WP and JP's "Build for App Store" lanes end up looking exactly the same, to unify them into a single lane that takes the `app` as an `option` for picking the right content just for the few app-specific values that those common calls will need to adapt to.
    - You could for example take some inspiration [on how this is done in WPAndroid](https://github.com/wordpress-mobile/WordPress-Android/blob/2b09cbe384623ccd67400c0a8dc71fb62f084ff2/fastlane/Fastfile#L3-L24), with the `APP_SPECIFIC_VALUES` constant defining all we need at the top, then have [a `get_app_name_option!` helper method](https://github.com/wordpress-mobile/WordPress-Android/blob/2b09cbe384623ccd67400c0a8dc71fb62f084ff2/fastlane/lanes/release.rb#L339-L346), and use that to [pick the right values from that `APP_SPECIFIC_VALUES` constant like this](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/fastlane/lanes/build.rb#L94-L96)


Obviously, all this is totally unrelated to this PR and should be done in a separate and subsequent PR. But I figured that it would be a good quick win and nice DRY-ing to do to simplify the lanes and avoid copy/pastes and logic duplication between WP and JP.